src/predictive-text-input/predictive-text-input.tsx

```diff
@@ -0,0 +1,222 @@
+import debounce from "lodash/debounce";
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { Input } from "../input";
+import { DropdownList } from "../shared/dropdown-list/dropdown-list";
+import { DropdownWrapper } from "../shared/dropdown-wrapper";
+import {
+    Divider,
+    SelectorDiv,
+} from "../shared/dropdown-wrapper/dropdown-wrapper.styles";
+import { PredictiveTextInputProps } from "./types";
+import { ItemsLoadStateType } from "../shared/dropdown-list/types";
+
+export const PredictiveTextInput = <T, V>({
+    className,
+    "data-testid": testId,
+    selectedOption,
+    minimumCharacters = 3,
+    fetchOptions,
+    placeholder = "Enter here...",
+    readOnly = false,
+    disabled = false,
+    error,
+    valueExtractor,
+    listExtractor,
+    displayValueExtractor,
+    onSelectOption,
+}: PredictiveTextInputProps<T, V>) => {
+    // =============================================================================
+    // CONST, STATE
+    // =============================================================================
+    const inputValue = selectedOption && displayValueExtractor(selectedOption);
+    const [input, setInput] = useState<string>(inputValue);
```

Comment:
default inputValue to `""` to avoid React's uncontrolled input warnings

Reply:
sorry, could have been more specific. initialising `input` based on the initial selectedOption is fine, but in the case where there's no selectedOption, `input` should be set to `""`